protocols/messages.go

```diff
@@ -1,19 +1,98 @@
 package protocols
 
 import (
+	"bytes"
 	"encoding/json"
+	"sort"
 
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/statechannels/go-nitro/channel/consensus_channel"
 	"github.com/statechannels/go-nitro/channel/state"
 	"github.com/statechannels/go-nitro/types"
 )
 
+const (
+	SignedStatePayload    PayloadType = "SignedStatePayload"
+	SignedProposalPayload PayloadType = "SignedProposalPayload"
+)
+
+type PayloadType string
+
 // Message is an object to be sent across the wire. It can contain a proposal and signed states, and is addressed to a counterparty.
 type Message struct {
-	To              types.Address
-	ObjectiveId     ObjectiveId
-	SignedStates    []state.SignedState
-	SignedProposals []consensus_channel.SignedProposal
+	To       types.Address
+	Payloads []MessagePayload
+}
+
+// MessagePayload is an objective id and EITHER a SignedState or SignedProposal.
+type MessagePayload struct {
+	ObjectiveId    ObjectiveId
+	SignedState    state.SignedState
+	SignedProposal consensus_channel.SignedProposal
+}
+
+// hasState returns true if the payload contains a signed state.
+func (p MessagePayload) hasState() bool {
+	stateHash, err := p.SignedState.State().Hash()
+	if err != nil {
+		return false
+	}
+
+	return bytes.Compare(stateHash.Bytes(), common.Hash{}.Bytes()) != 0
```

Comment:
Previously the `state` `Equal` function would fail when dealing with `nil` big.ints. This meant that `p.SignedState.State().Equal(state.State{})` would `panic`.

I've addressed that in https://github.com/statechannels/go-nitro/pull/582/commits/5084116ba72166ac3c7de1de4cf5d182e81d1f7b by updating `Equal` to handle `nil` big ints.